### PR TITLE
Gihub actions to run tests

### PR DESCRIPTION
**Pull request by jbygdell**

### CODE REVIEW

**Review by norling — APPROVED** [review]
I had planned to have the modules return their errors to be handled by main, but I see you added a `log.Fatal` to the argument parsing. Do you prefer to have the modules not return errors? Either is fine with me :)

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
```
Comment (aaperis) [inline .github/workflows/codeql-analysis.yml:5]:
```suggestion
    branches: [main]
```
not sure if this essential but still..

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
```
Comment (aaperis) [inline .github/workflows/codeql-analysis.yml:8]:
```suggestion
    branches: [main]
```
same as above

**Review by aaperis — APPROVED** [review]
looks good :-)

**Pull request merged**